src/library/basetracktablemodel.cpp

```diff
@@ -546,17 +553,21 @@ QVariant BaseTrackTableModel::composeCoverArtToolTipHtml(
     unsigned int absoluteHeightOfCoverartToolTip = static_cast<int>(
             pViewScreen->availableGeometry().height() *
             kRelativeHeightOfCoverartToolTip);
-    // Get image from cover art cache
-    CoverArtCache* pCache = CoverArtCache::instance();
-    QPixmap pixmap = QPixmap(absoluteHeightOfCoverartToolTip,
-            absoluteHeightOfCoverartToolTip); // Height also used as default for the width, in assumption that covers are squares
-    pixmap = pCache->tryLoadCover(this,
-            getCoverInfo(index),
-            absoluteHeightOfCoverartToolTip,
-            CoverArtCache::Loading::NoSignal);
+    const auto coverInfo = getCoverInfo(index);
+    if (!coverInfo.hasImage()) {
+        return QPixmap();
+    }
+    m_toolTipIndex = index;
+    QPixmap pixmap = CoverArtCache::getCachedCover(
+            coverInfo,
+            absoluteHeightOfCoverartToolTip);
     if (pixmap.isNull()) {
-        // Cache miss -> Don't show a tooltip
-        return QVariant();
+        // Cache miss -> Request Cover via coverFound signal
+        CoverArtCache::requestUncachedCover(
+                this,
+                coverInfo,
+                absoluteHeightOfCoverartToolTip);
+        return tr("Fetching image ...");
```

Comment:
What is the string used for? Can you add a comment?

Reply:
This text is shown as tooltip for the spit of a second before the actual cover is shown. I will add a comment.

Reply:
Done

Reply:
> This text is shown as tooltip for the spit of a second before the actual cover is shown.

Than it doesn't work, I see the text, but it's no replaced by the cover art. If I don't move the mouse, it remains there for some seconds, and than the tooltip disappears.